plugin/pkg/scheduler/core/scheduling_queue.go

```diff
@@ -337,21 +341,50 @@ func (p *PriorityQueue) AssignedPodUpdated(pod *v1.Pod) {
 // function adds all pods and then signals the condition variable to ensure that
 // if Pop() is waiting for an item, it receives it after all the pods are in the
 // queue and the head is the highest priority pod.
-// TODO(bsalamat): We should add a back-off mechanism here so that a high priority
-// pod which is unschedulable does not go to the head of the queue frequently. For
-// example in a cluster where a lot of pods being deleted, such a high priority
-// pod can deprive other pods from getting scheduled.
+// We add a back-off mechanism here so that a high priority pod which is
+// unschedulable does not go to the head of the queue frequently. For example
+// in a cluster where a lot of pods being deleted, such a high priority pod
+// can deprive other pods from getting scheduled.
 func (p *PriorityQueue) MoveAllToActiveQueue() {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 	var unschedulablePods []interface{}
+
+	p.podBackoff.Gc()
+
 	for _, pod := range p.unschedulableQ.pods {
-		unschedulablePods = append(unschedulablePods, pod)
+		podID := ktypes.NamespacedName{
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
+		}
+		_, ok := p.podBackoff.GetPerPodBackoff()[podID]
+		if !ok {
+			unschedulablePods = append(unschedulablePods, pod)
+			p.podBackoff.GetEntry(podID)
+		} else {
+			go func() {
```

Comment:
The `for` loop can start many go routines that call activeQ and unschedulableQ functions without any synchronization. This will be problematic.